From 0162fcba342d5d13887080d1451b859c0339f42b Mon Sep 17 00:00:00 2001
From: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
Date: Mon, 14 Mar 2022 12:47:08 +0800
Subject: [PATCH 31/54] configs: arm64: imx8m{m,n}: Change to 1:1 IPA-PA
 mapping

No SMMU implemented on these 2 platforms, so the current non 1:1 mapping
will result in the DMAs access wrong physical address.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
---
 configs/arm64/imx8mm-harpoon-freertos-audio.c      | 4 ++--
 configs/arm64/imx8mm-harpoon-freertos-industrial.c | 4 ++--
 configs/arm64/imx8mm-harpoon-freertos.c            | 4 ++--
 configs/arm64/imx8mn-harpoon-freertos-audio.c      | 4 ++--
 configs/arm64/imx8mn-harpoon-freertos-industrial.c | 4 ++--
 configs/arm64/imx8mn-harpoon-freertos.c            | 4 ++--
 6 files changed, 12 insertions(+), 12 deletions(-)

diff --git a/configs/arm64/imx8mm-harpoon-freertos-audio.c b/configs/arm64/imx8mm-harpoon-freertos-audio.c
index 29bc5a13..864fb4a2 100644
--- a/configs/arm64/imx8mm-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mm-harpoon-freertos-audio.c
@@ -17,12 +17,12 @@
 /* Name, cores, entry point */
 #define CONFIG_INMATE_NAME		"freertos"
 #define CONFIG_INMATE_CORE_BITMAP	(0b1000)
-#define CONFIG_INMATE_BASE		(0xc0000000)
+#define CONFIG_INMATE_BASE		(0x93c00000)
 
 /* Memory & peripherals */
 #define CONFIG_INMATE_REGIONS		\
 	RTOS_COMMON_MEMORY_REGIONS,                                     \
-	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */    \
+	MEM_REGION_RWXL(0x93c00000, 0x93c00000, MB(16)),   /* RAM */    \
 	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
 	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
diff --git a/configs/arm64/imx8mm-harpoon-freertos-industrial.c b/configs/arm64/imx8mm-harpoon-freertos-industrial.c
index dbe7fda1..cda28541 100644
--- a/configs/arm64/imx8mm-harpoon-freertos-industrial.c
+++ b/configs/arm64/imx8mm-harpoon-freertos-industrial.c
@@ -16,12 +16,12 @@
 /* Name, cores, entry point */
 #define CONFIG_INMATE_NAME		"freertos"
 #define CONFIG_INMATE_CORE_BITMAP	(0b1000)
-#define CONFIG_INMATE_BASE		(0xc0000000)
+#define CONFIG_INMATE_BASE		(0x93c00000)
 
 /* Memory & peripherals */
 #define CONFIG_INMATE_REGIONS		\
 	RTOS_COMMON_MEMORY_REGIONS,                                     \
-	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */    \
+	MEM_REGION_RWXL(0x93c00000, 0x93c00000, MB(16)),   /* RAM */    \
 	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
 	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
diff --git a/configs/arm64/imx8mm-harpoon-freertos.c b/configs/arm64/imx8mm-harpoon-freertos.c
index 688155d1..b3a62670 100644
--- a/configs/arm64/imx8mm-harpoon-freertos.c
+++ b/configs/arm64/imx8mm-harpoon-freertos.c
@@ -17,12 +17,12 @@
 /* Name, cores, entry point */
 #define CONFIG_INMATE_NAME		"freertos"
 #define CONFIG_INMATE_CORE_BITMAP	(0b1000)
-#define CONFIG_INMATE_BASE		(0xc0000000)
+#define CONFIG_INMATE_BASE		(0x93c00000)
 
 /* Memory & peripherals */
 #define CONFIG_INMATE_REGIONS		\
 	RTOS_COMMON_MEMORY_REGIONS,                                    \
-	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */   \
+	MEM_REGION_RWXL(0x93c00000, 0x93c00000, MB(16)),   /* RAM */    \
 	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */ \
 	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */  \
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128))   /* ITCM */
diff --git a/configs/arm64/imx8mn-harpoon-freertos-audio.c b/configs/arm64/imx8mn-harpoon-freertos-audio.c
index 9d9898aa..f099471c 100644
--- a/configs/arm64/imx8mn-harpoon-freertos-audio.c
+++ b/configs/arm64/imx8mn-harpoon-freertos-audio.c
@@ -17,12 +17,12 @@
 /* Name, cores, entry point */
 #define CONFIG_INMATE_NAME		"freertos"
 #define CONFIG_INMATE_CORE_BITMAP	(0b1000)
-#define CONFIG_INMATE_BASE		(0xc0000000)
+#define CONFIG_INMATE_BASE		(0x93c00000)
 
 /* Memory & peripherals */
 #define CONFIG_INMATE_REGIONS		\
 	RTOS_COMMON_MEMORY_REGIONS,                                     \
-	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */    \
+	MEM_REGION_RWXL(0x93c00000, 0x93c00000, MB(16)),   /* RAM */    \
 	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
 	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
diff --git a/configs/arm64/imx8mn-harpoon-freertos-industrial.c b/configs/arm64/imx8mn-harpoon-freertos-industrial.c
index e514dbd8..39dfc4e8 100644
--- a/configs/arm64/imx8mn-harpoon-freertos-industrial.c
+++ b/configs/arm64/imx8mn-harpoon-freertos-industrial.c
@@ -16,12 +16,12 @@
 /* Name, cores, entry point */
 #define CONFIG_INMATE_NAME		"freertos"
 #define CONFIG_INMATE_CORE_BITMAP	(0b1000)
-#define CONFIG_INMATE_BASE		(0xc0000000)
+#define CONFIG_INMATE_BASE		(0x93c00000)
 
 /* Memory & peripherals */
 #define CONFIG_INMATE_REGIONS		\
 	RTOS_COMMON_MEMORY_REGIONS,                                     \
-	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */    \
+	MEM_REGION_RWXL(0x93c00000, 0x93c00000, MB(16)),   /* RAM */    \
 	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */  \
 	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */   \
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128)),  /* ITCM */   \
diff --git a/configs/arm64/imx8mn-harpoon-freertos.c b/configs/arm64/imx8mn-harpoon-freertos.c
index a477f6d5..ae290791 100644
--- a/configs/arm64/imx8mn-harpoon-freertos.c
+++ b/configs/arm64/imx8mn-harpoon-freertos.c
@@ -17,12 +17,12 @@
 /* Name, cores, entry point */
 #define CONFIG_INMATE_NAME		"freertos"
 #define CONFIG_INMATE_CORE_BITMAP	(0b1000)
-#define CONFIG_INMATE_BASE		(0xc0000000)
+#define CONFIG_INMATE_BASE		(0x93c00000)
 
 /* Memory & peripherals */
 #define CONFIG_INMATE_REGIONS		\
 	RTOS_COMMON_MEMORY_REGIONS,                                    \
-	MEM_REGION_RWXL(0x93c00000, 0xc0000000, MB(16)),   /* RAM */   \
+	MEM_REGION_RWXL(0x93c00000, 0x93c00000, MB(16)),   /* RAM */    \
 	MEM_REGION_RWX( 0x00900000, 0x00900000, KB(384)),  /* OCRAM */ \
 	MEM_REGION_RW(  0x00800000, 0x00800000, KB(128)),  /* DTCM */  \
 	MEM_REGION_RWX( 0x007e0000, 0x007e0000, KB(128))   /* ITCM */
-- 
2.34.1

